### PR TITLE
Add missing outputs to apm server spec file

### PR DESCRIPTION
**Pull request by urso**
- Bug

## What does this PR do?

- Add missing outputs to apm-server spec file.

## Why is it important?

Parity with other spec files for services based on libbeat.

### CODE REVIEW

File: x-pack/elastic-agent/spec/apm-server.yml [hunk 1]
```diff
@@ -32,4 +32,5 @@ rules:
         - output
         - fleet
   - inject_headers: {}
-when: length(${inputs}) > 0 and hasKey(${output}, 'elasticsearch')
+when: length(${inputs}) > 0 and hasKey(${output}, 'elasticsearch', 'redis',
+  'kafka', 'logstash')
```
Comment (simitt) [inline x-pack/elastic-agent/spec/apm-server.yml:36]: My information was that `redis` will not be supported for now?

Comment (urso) [reply x-pack/elastic-agent/spec/apm-server.yml:36]: just copy and paste from the other Beats. I would like to keep all libbeat based process types in sync. Removing redis should be done separately.

**Review by michalpristas — APPROVED** [review]
please add changelog, lgtm
silvia has some copy rules with elasticsearch hardcoded (not in just yet), we need to verify later on if this wont' fail if ES is missing

**Comment by urso** [comment]
>  silvia has some copy rules with elasticsearch hardcoded (not in just yet), we need to verify later on if this wont' fail if ES is missing

This might be a problem in general. The condition is only executed at the end. If users have configured LS, that might still fail even before we hit the condition.

**Comment by michalpristas** [comment]
> This might be a problem in general. The condition is only executed at the end. If users have configured LS, that might still fail even before we hit the condition.
yeah we just need to make sure it wont panics. if it's missing it should be noop.

**Pull request merged**